org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AbstractMatcher.java

```diff
@@ -158,11 +158,21 @@ final void next() {
 	 * {@link AbstractInsnNode#LABEL}, {@link AbstractInsnNode#LINE}.
 	 */
 	final void skipNonOpcodes() {
+		cursor = skipNonOpcodes(cursor);
+	}
+
+	/**
+	 * Returns first instruction from given and following it that is not
+	 * {@link AbstractInsnNode#FRAME}, {@link AbstractInsnNode#LABEL},
+	 * {@link AbstractInsnNode#LINE}.
+	 */
+	final AbstractInsnNode skipNonOpcodes(AbstractInsnNode cursor) {
```

Comment:
As this method does not change state (unlike the others) I would prefer to declare it `static`.

Reply:
Indeed. Fixed!